packages/core/src/virtual-test-entrypoint.ts

```diff
@@ -0,0 +1,201 @@
+import { AppFiles } from './app-files';
+import { compile } from './js-handlebars';
+import type { Resolver } from './module-resolver';
+import { join } from 'path';
+import { extensionsPattern } from '@embroider/shared-internals';
+import walkSync from 'walk-sync';
+import type { V2AddonPackage } from '@embroider/shared-internals/src/package';
+import { staticAppPathsPattern } from './virtual-entrypoint';
+
+const entrypointPattern = /(?<filename>.*)[\\/]-embroider-test-entrypoint.js/;
+
+export function decodeTestEntrypoint(filename: string): { fromFile: string } | undefined {
+  // Performance: avoid paying regex exec cost unless needed
+  if (!filename.includes('-embroider-test-entrypoint.js')) {
+    return;
+  }
+  let m = entrypointPattern.exec(filename);
+  if (m) {
+    return {
+      fromFile: m.groups!.filename,
+    };
+  }
+}
+
+export function renderTestEntrypoint(
+  resolver: Resolver,
+  { fromFile }: { fromFile: string }
+): { src: string; watches: string[] } {
+  const owner = resolver.packageCache.ownerOfFile(fromFile);
+
+  if (!owner) {
+    throw new Error('Owner expected');
+  }
+
+  let engine = resolver.owningEngine(owner);
+  let hasFastboot = Boolean(resolver.options.engines[0]!.activeAddons.find(a => a.name === 'ember-cli-fastboot'));
+
+  let appFiles = new AppFiles(
+    {
+      package: owner,
+      addons: new Map(
+        engine.activeAddons.map(addon => [
+          resolver.packageCache.get(addon.root) as V2AddonPackage,
+          addon.canResolveFromFile,
+        ])
+      ),
+      isApp: true,
+      modulePrefix: resolver.options.modulePrefix,
+      appRelativePath: 'NOT_USED_DELETE_ME',
+    },
+    getAppFiles(owner.root),
+    hasFastboot ? getFastbootFiles(owner.root) : new Set(),
+    extensionsPattern(resolver.options.resolvableExtensions),
+    staticAppPathsPattern(resolver.options.staticAppPaths),
+    resolver.options.podModulePrefix
+  );
+
+  let amdModules: { runtime: string; buildtime: string }[] = [];
+
+  for (let relativePath of appFiles.tests) {
+    amdModules.push(importPaths(resolver, appFiles, relativePath));
+  }
+
+  let src = entryTemplate({
+    amdModules,
+    testSuffix: true,
+    // this is a backward-compatibility feature: addons can force inclusion of test support modules.
+    defineModulesFrom: './-embroider-implicit-test-modules.js',
```

Comment:
Should the tests entrypoint be doing this? Remember that it imports the app entrypoint, so anything the app entrypoint is doing shouldn't be repeated in here.

Reply:
you're right this shouldn't be needed, it's already importing the app entrypoint and that is handled there 👍 removed